src/google/TreeMap.js

```diff
@@ -0,0 +1,158 @@
+    
+    TreeMap.prototype.publish("noColor", null, "html-color", "The color to use for a rectangle when a node has no value for column 3, and that node is a leaf (or contains only leaves). Specify an HTML color value.",{tags:['Basic']});
+    TreeMap.prototype.publish("noHighlightColor", null, "html-color", "The color to use for a rectangle of 'no' color when highlighted. Specify an HTML color value or null; if null, this will be the value of noColor lightened by 35%.",null,{tags:['Basic']});
+    
+    TreeMap.prototype.publish("showScale", true, "boolean", "Whether or not to show a color gradient scale from minColor to maxColor along the top of the chart. Specify true to show the scale.",null,{tags:['Intermediate']});
+
+    TreeMap.prototype.publish("showTooltips", true, "boolean", "Whether or not to show tooltips.",null,{tags:['Basic']});
+    
+    TreeMap.prototype.publish("useWeightedAverageForAggregation", true, "boolean", "Whether to use weighted averages for aggregation.",null,{tags:['Basic']});
+    
+    TreeMap.prototype.getChartOptions = function () {
+        var retVal = [];
+        
+        retVal.headerColor = this.headerColor();
+        retVal.headerHeight = this.headerHeight();
+        retVal.headerHighlightColor = this.headerHighlightColor();
+        retVal.hintOpacity = this.hintOpacity();
+        retVal.maxColor = this.maxColor();
+        retVal.maxDepth = this.maxDepth();
+        retVal.maxHighlightColor = this.maxHighlightColor();
+        retVal.maxPostDepth = this.maxPostDepth();
+        retVal.maxColorValue = this.maxColorValue();
+        retVal.midColor = this.midColor();
+        retVal.midHighlightColor = this.midHighlightColor();
+        retVal.minColor = this.minColor();
+        retVal.minHighlightColor = this.minHighlightColor();
+        retVal.minColorValue = this.minColorValue();
+        retVal.noColor = this.noColor();
+        retVal.noHighlightColor = this.noHighlightColor();
+        retVal.showScale = this.showScale();
+        retVal.showTooltips = this.showTooltips();
+        retVal.useWeightedAverageForAggregation = this.useWeightedAverageForAggregation();
+        retVal.generateTooltip = TreeMap.prototype.defaultlTooltip;
+        return retVal;
+    };
+
+    TreeMap.prototype.enter = function (domNode, element) {
+        element.style("overflow", "hidden");
+        HTMLWidget.prototype.enter.apply(this, arguments);
+        this.treemapChart = new google.visualization.TreeMap(element.node());
+    };
+
+    TreeMap.prototype.update = function (domNode, element) {
+        HTMLWidget.prototype.update.apply(this, arguments);
+        TreeMap.prototype.mapData = this._data_google;
+        this.treemapChart.draw(this._data_google, this.getChartOptions());
+    };
+    
+    TreeMap.prototype.data = function (_) {
+        if (arguments.length) {
```

Comment:
See above comment(s) re data.  Typically all setters/getters will return the current value if no arguments are passed in:
So ```var data = this.data();``` should work as expected...